tests/python/gaia-ui-tests/gaiatest/tests/functional/cost_control/test_cost_control_data_alert_mobile.py

```diff
@@ -46,29 +41,19 @@ def test_cost_control_data_alert_mobile(self):
         search = Search(self.marionette)
         search.launch(launch_timeout=30000)
         browser = search.go_to_url('http://mozqa.com/qa-testcase-data/Images/sample_png_02.png')
-        browser.wait_for_page_to_load(180)
-
-        browser.switch_to_content()
-        Wait(self.marionette, timeout=60).until(lambda m: "sample_png_02.png" in m.title)
-        browser.switch_to_chrome()
+        browser.wait_for_page_to_load(timeout=180)
+        browser.wait_for_title_to_load(title='sample_png_02.png')
 
         system = System(self.marionette)
+        system.wait_for_notification_toaster_displayed()
+        system.wait_for_notification_toaster_not_displayed()
         utility_tray = system.open_utility_tray()
         utility_tray.wait_for_notification_container_displayed()
 
-        # switch to cost control widget
-        usage_iframe = self.marionette.find_element(*self._cost_control_widget_locator)
-        self.marionette.switch_to_frame(usage_iframe)
-
-        # make sure the color changed
-        # The timeout is increased, because for some reason, it takes some time
-        # before the limit view is shown (the browser has to finish loading?)
-        usage_view = self.marionette.find_element(*self._data_usage_view_locator)
-        Wait(self.marionette, timeout=40).until(lambda m: 'reached-limit' in usage_view.get_attribute('class'),
-             message='Data usage bar did not breach limit')
-        usage_view.tap()
-
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == cost_control.name)
+        cost_control_widget = utility_tray.cost_control_widget
+        cost_control_widget.wait_for_limit_to_be_reached()
```

Comment:
It is strange that we have to wait for the limit to be reached here, when the notification toaster about 'limit reached' has already been shown.
But it's better to be safe. Iirc, this test is really fragile.